src/main/java/org/sagebionetworks/bridge/services/ActivityEventService.java

```diff
@@ -147,9 +165,58 @@ public void publishActivityEvent(ActivityEvent event) {
      */
     public Map<String, DateTime> getActivityEventMap(String healthCode) {
         checkNotNull(healthCode);
+        
         return activityEventDao.getActivityEventMap(healthCode);
     }
 
+    /**
+     * Gets the activity events times for a specific user in order to schedule against them.
+     */
+    public Map<String, DateTime> getActivityEventMap(Account account) {
+        checkNotNull(account.getHealthCode());
+        Map<String, DateTime> activityMap = activityEventDao.getActivityEventMap(account.getHealthCode());
+        Builder<String, DateTime> builder = ImmutableMap.<String, DateTime>builder();
+        
+        DateTime studyStartDateTime = activityMap.get(ActivityEventObjectType.STUDY_START_DATE.name().toLowerCase());
+        DateTime activitiesRetrievedDateTime = activityMap.get(ActivityEventObjectType.ACTIVITIES_RETRIEVED.name().toLowerCase());
+        DateTime enrollmentDateTime = activityMap.get(ActivityEventObjectType.ENROLLMENT.name().toLowerCase());
+        
+        if (activitiesRetrievedDateTime != null) {
+            studyStartDateTime = activitiesRetrievedDateTime;
+        } else if (enrollmentDateTime != null) {
+            studyStartDateTime = enrollmentDateTime;
+        } else if (studyStartDateTime == null) {
+            Study study = studyService.getStudy(account.getStudyId());
+            StudyParticipant studyParticipant = participantService.getParticipant(study, account.getHealthCode());
+            studyStartDateTime = studyParticipant.getCreatedOn();
+            builder.put(ActivityEventObjectType.STUDY_START_DATE.name().toLowerCase(), studyStartDateTime);
```

Comment:
Note that the first 2 branches don't actually save the study_start_date anywhere.